### PR TITLE
Implementation of the device factory handler

### PR DESCRIPTION
**Pull request by tcalmant**
Implements the device factory handler.
Requires #81 to be merged to work.

### CODE REVIEW

File: prototype/southbound/device-factory/device-factory-core/src/main/java/org/eclipse/sensinact/gateway/southbound/device/factory/impl/FactoryParserHandler.java [hunk 1]
```diff
@@ -0,0 +1,709 @@
...
+    }
+
+    /**
+     * Looks for a location in the given record
+     *
+     * @param record       Record to read
+     * @param placeholders Defined mapping placeholders
+     * @return The parsed location as a GeoJSON string or null
+     */
+    private String computeLocation(final IDeviceMappingRecord record, final Map<String, RecordPath> placeholders) {
```
Comment (timothyjward) [inline prototype/southbound/device-factory/device-factory-core/src/main/java/org/eclipse/sensinact/gateway/southbound/device/factory/impl/FactoryParserHandler.java:410]: I think that this should be returning a GeoJson Object, not a String. That's what the value of the admin/location should be.

File: prototype/southbound/device-factory/device-factory-core/src/main/java/org/eclipse/sensinact/gateway/southbound/device/factory/impl/FactoryParserHandler.java [hunk 2]
```diff
@@ -0,0 +1,709 @@
...
+
+        if (svcRefs == null || svcRefs.isEmpty()) {
+            throw new MissingParserException(String.format("Parser ID '%s' is missing", parserId));
+        }
+
+        return svcRefs.stream().sorted(Comparator.naturalOrder()).findFirst().get();
+    }
+
+    @Override
+    public void handle(final SensiNactSession session, final DeviceMappingConfigurationDTO configuration,
```
Comment (timothyjward) [inline prototype/southbound/device-factory/device-factory-core/src/main/java/org/eclipse/sensinact/gateway/southbound/device/factory/impl/FactoryParserHandler.java:144]: I think that the `SensiNactSession` can be removed from here now? If not then what user does it act as?

**Review by timothyjward — APPROVED** [review]

File: prototype/southbound/device-factory/device-factory-core/src/main/java/org/eclipse/sensinact/gateway/southbound/device/factory/impl/FactoryParserHandler.java (hunk 1 above)
Comment (tcalmant) [reply prototype/southbound/device-factory/device-factory-core/src/main/java/org/eclipse/sensinact/gateway/southbound/device/factory/impl/FactoryParserHandler.java:410]: Agreed, will do

File: prototype/southbound/device-factory/device-factory-core/src/main/java/org/eclipse/sensinact/gateway/southbound/device/factory/impl/FactoryParserHandler.java (hunk 2 above)
Comment (tcalmant) [reply prototype/southbound/device-factory/device-factory-core/src/main/java/org/eclipse/sensinact/gateway/southbound/device/factory/impl/FactoryParserHandler.java:144]: Agreed

**Pull request merged**